todo/todo/src/App.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react';
+import Todos from './components/Todos';
+import TodoForm from './components/TodoForm';
+import { connect } from 'react-redux';
+import { addTodo } from './actions';
+
+class App extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            todoText: ''
+        };
+    }
+
+    handleInputChange = e => {
+        this.setState({ [e.target.name]: e.target.value });
+    };
+
+    addTodo = () => {
+        const { todoText } = this.state;
+        const newTodo = {
+            id: this.props.todos.length + 1,
+            completed: false,
+            text: todoText
+        };
+        this.props.addTodo(newTodo);
+        this.setState({ todoText: '' });
+    };
+
+    render() {
+        return (
+            <div className="App">
+                <TodoForm
+                    handleInputChange={this.handleInputChange}
+                    todoText={this.state.todoText}
+                    addTodo={this.addTodo}
```

Comment:
Instead of passing these down as props, you should create these in you `TodoForm` component. It would clean up the code and make things a bit more organized.